### PR TITLE
feat: add contact enabled setting

### PR DESCRIPTION
**Pull request by dolcalmi**
Split PR according to https://github.com/GaloyMoney/galoy/pull/1691#issuecomment-1254949049

### CODE REVIEW

File: src/app/accounts/create-account.ts [hunk 1]
```diff
@@ -50,6 +50,7 @@ const setupAccount = async ({
   // FIXME: to remove when Casbin is been introduced
   const role = getTestAccounts().find(({ phone }) => phone === phoneNumberValid)?.role
   account.role = role
+  account.contactEnabled = !account.role || account.role === "user"
```
Comment (nicolasburtey) [inline src/app/accounts/create-account.ts:53]: I'm not sure this is very explicit on the intent: `!account.role`

File: src/app/payments/send-intraledger.ts [hunk 2]
```diff
@@ -281,28 +279,24 @@ const executePaymentViaIntraledger = async ({
...
 }: {
-  senderAccountId: AccountId
-  senderUsername: Username | undefined
-  recipientAccountId: AccountId
-  recipientUsername: Username | undefined
+  senderAccount: Account
+  recipientAccount: Account
 }): Promise<true | ApplicationError> => {
-  if (recipientUsername) {
+  if (senderAccount.contactEnabled && recipientAccount.username) {
```
Comment (nicolasburtey) [inline src/app/payments/send-intraledger.ts:288]: I'm wondering it the use is more something like:

```
if (!(senderAccount.contactEnabled && recipientAccount.contactEnabled)) {
   return
}

continue...
```

ie: only add contact if both contact have contactEnabled

in the case of the quiz, I think it doens't make sense to have the funder add contacts, but also doesn't make sense to have the end user have the contact of the funder. 

wdyt?

File: src/migrations/20220922030802-update-contact-enabled.ts [hunk 3]
```diff
@@ -0,0 +1,18 @@
+module.exports = {
+  async up(db) {
+    try {
+      const collection = db.collection("users")
+      const result = await collection.updateMany(
+        { role: { $ne: "user" } },
```
Comment (nicolasburtey) [inline src/migrations/20220922030802-update-contact-enabled.ts:6]: maybe editor could be excluded also.

File: src/app/accounts/create-account.ts (hunk 1 above)
Comment (vindard) [reply src/app/accounts/create-account.ts:53]: Based on [this comment](https://github.com/GaloyMoney/galoy/pull/1696/files#r979055348) should `editor` be included here too?

File: test/helpers/user.ts [hunk 4]
```diff
@@ -191,7 +191,11 @@ export const createUserAndWallet = async (entry: TestEntry) => {
   }
 
   if (entry.role) {
-    await User.findOneAndUpdate({ _id: toObjectId<UserId>(uid) }, { role: entry.role })
+    const contactEnabled = entry.role === "user"
```
Comment (vindard) [inline test/helpers/user.ts:194]: Same comment here

**Review by vindard — COMMENTED** [review]
Just one small thing based on Nicolas's comments

File: test/helpers/user.ts (hunk 4 above)
Comment (dolcalmi) [reply test/helpers/user.ts]: fixed https://github.com/GaloyMoney/galoy/pull/1696/commits/ab3bd473794af580a5739e38e2862bf15a03b806

File: src/app/accounts/create-account.ts (hunk 1 above)
Comment (dolcalmi) [reply src/app/accounts/create-account.ts]: fixed https://github.com/GaloyMoney/galoy/pull/1696/commits/ab3bd473794af580a5739e38e2862bf15a03b806

File: src/app/payments/send-intraledger.ts (hunk 2 above)
Comment (dolcalmi) [reply src/app/payments/send-intraledger.ts]: makes sense.. updated here https://github.com/GaloyMoney/galoy/pull/1696/commits/b35293a215a997ad35b6583d001bbbd8b5bfa8c5

File: src/migrations/20220922030802-update-contact-enabled.ts (hunk 3 above)
Comment (dolcalmi) [reply src/migrations/20220922030802-update-contact-enabled.ts]: Fixed https://github.com/GaloyMoney/galoy/pull/1696/commits/ab3bd473794af580a5739e38e2862bf15a03b806

**Review by vindard — COMMENTED** [review]
Looks good to me, just one small question

File: src/migrations/20220922030802-update-contact-enabled.ts [hunk 5]
```diff
@@ -0,0 +1,18 @@
+module.exports = {
+  async up(db) {
+    try {
+      const collection = db.collection("users")
+      const result = await collection.updateMany(
+        { role: { $in: ["dealer", "funder", "bankowner"] } },
+        { $set: { contactEnabled: false, contacts: [] } },
```
[the remark this answers is not in the document]
Comment (dolcalmi) [reply src/migrations/20220922030802-update-contact-enabled.ts:7]: schema default value is true so there is no need to do the update

**Review by vindard — APPROVED** [review]

**Pull request merged**